File: CodingSamples/DotNet/Advanced/Networking/TcpServerTest/DemoApp1/Models/ShopModel.cs
using System.Text.Json;

namespace DemoApp.Models;

public class ShopModel
{
    private IEnumerable<ItemInfo> items;

    public ShopModel()
    {
        items = Load("EviTek.store");
    }

    public ItemInfo GetItemInfo(string name)
    {
        return items.FirstOrDefault(i => i.Id == name);
    }

    public static void Save(string document, ItemInfo[] entries)
    {
        using var output = new FileStream(document, FileMode.Create);
        //serialization - convverting objects into stream of bytes
        JsonSerializer.Serialize(output, entries);
    }

    public static ItemInfo[] Load(string document)
    {
        using var input = new FileStream(document, FileMode.Open);
        //deserialization - converting stream of bytes into objects
        return JsonSerializer.Deserialize<ItemInfo[]>(input);
    }
}

File: CodingSamples/Java/Networking/TCPServerTest/demo/src/main/java/app/models/ShopModel.java
package app.models;

import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.util.stream.Stream;

import com.fasterxml.jackson.databind.ObjectMapper;

public class ShopModel {

    private ItemInfo[] items;

    public ShopModel() {
        items = load("EviTek.store");
    }

    public ItemInfo getItemInfo(String name) {
        return Stream.of(items)
            .filter(i -> i.id().equals(name))
            .findFirst()
            .orElse(null);
    }
    
    public static void save(String document, ItemInfo[] entries) {
        var serializer = new ObjectMapper();
        try(var output = new FileOutputStream(document)){
            serializer.writeValue(output, entries);
        }catch(IOException e){
            throw new RuntimeException(e);
        }
    }

    public static ItemInfo[] load(String document) {
        var serializer = new ObjectMapper();
        try(var input = new FileInputStream(document)){
            return serializer.readValue(input, ItemInfo[].class);
        }catch(IOException e){
            throw new RuntimeException(e);
        }
    }
}
